node/node.go

```diff
@@ -342,50 +352,52 @@ func (n *Node) startRPC() error {
 	if err := n.startInProc(); err != nil {
```

Comment:
I think it's cleaner to easier to rename this function `startExternalRPC`, remove `n.startInProc()` from it entirely, and avoid extra layer of branching on `!n.readOnly`.

Overall: You have a lot of `if !n.readOnly` branches, which is a bad code smell -- try to refactor the code so that the read-only special case is self-contained.